### PR TITLE
[pt2] enable meta tests for `foreach` ops

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #107566

Rebased #107566 and fixed some skips.

cc @mruberry @ZainRizvi

### CODE REVIEW

File: test/test_meta.py [hunk 1]
```diff
@@ -1086,6 +1097,20 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
 
         return expected
 
+
+def _handle_foreach_kwargs(func, args, kwargs):
+    if 'zero_size' in kwargs:
+        kwargs.pop('zero_size')
+    if 'disable_fastpath' in kwargs:
+        kwargs.pop('disable_fastpath')
```
Comment (lezcano) [inline test/test_meta.py:1105]: Why are these sampled in the first place?

File: test/test_meta.py [hunk 2]
```diff
@@ -1086,6 +1097,20 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
...
+    if func in (torch._foreach_addcmul, torch._foreach_addcmul_, torch._foreach_addcdiv, torch._foreach_addcdiv_):
+        if 'values' in kwargs:
+            values = kwargs.pop("values")
+            if values is not None:
+                args = list(args)
+                args.append(values)
```
Comment (lezcano) [inline test/test_meta.py:1111]: Probably rather than doing this, we may want to have separate sample functions for these 4 that sample the correct inputs

**Review by lezcano — COMMENTED** [review]
I don't like the `_handle_foreach_kwargs` pattern. It'd be better to just sample the correct samples in the first place.

File: test/test_meta.py (hunk 1 above)
Comment (isuruf) [reply test/test_meta.py:1105]: I'm not sure about the `zero_size` kwarg. Removing it still makes the tests pass.

About `disable_fastpath`, it's a keyword argument for testing with a fastpath and is not a keyword argument to the operation itself. I guess we could have a separate op_db with a value of `with_fastpath=True` and use that in tests that uses the fastpath argument.

**Comment by isuruf** [comment]
Closing in favour of #112280

**Pull request closed without merging**